rules/terraformrules/terraform_empty_list_equality.go

```diff
@@ -0,0 +1,88 @@
+package terraformrules
+
+import (
+	"log"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/hclsyntax"
+	"github.com/terraform-linters/tflint/tflint"
+)
+
+// TerraformEmptyListCheckRule checks whether is there a comparison with an empty list
+type TerraformEmptyListCheckRule struct{}
+
+// NewTerraformCommentSyntaxRule returns a new rule
+func NewTerraformEmptyListCheckRule() *TerraformEmptyListCheckRule {
+	return &TerraformEmptyListCheckRule{}
+}
+
+// Name returns the rule name
+func (r *TerraformEmptyListCheckRule) Name() string {
+	return "terraform_empty_list_equality"
+}
+
+// Enabled returns whether the rule is enabled by default
+func (r *TerraformEmptyListCheckRule) Enabled() bool {
+	return true
+}
+
+// Severity returns the rule severity
+func (r *TerraformEmptyListCheckRule) Severity() tflint.Severity {
+	return tflint.WARNING
+}
+
+// Link returns the rule reference link
+func (r *TerraformEmptyListCheckRule) Link() string {
+	return tflint.ReferenceLink(r.Name())
+}
+
+// Check checks whether the list is being compared with static empty list
+func (r *TerraformEmptyListCheckRule) Check(runner *tflint.Runner) error {
+	if !runner.TFConfig.Path.IsRoot() {
+		// This rule does not evaluate child modules.
+		return nil
+	}
+
+	log.Printf("[TRACE] Check `%s` rule for `%s` runner", r.Name(), runner.TFConfigPath())
+
+	files := make(map[string]*struct{})
+	for _, variable := range runner.TFConfig.Module.Variables {
```

Comment:
Not sure how this relates, i.e. iterating through all variable declarations and using those filenames. Doesn't seem like that's needed?

Reply:
again, bad copy pasta from a different rule. thank you